### PR TITLE
SETI-1030: Trigger check pipeline in zuul when change PR base

### PR DESCRIPTION
**Pull request by janpriessnitz**

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/zuul-opensource-repository-check/72/ : SUCCESS in 21s
- https://ci.intgdc.com/job/zuul-tox-zuul-el7/84/ : FAILURE in 10m 44s

**Comment by janpriessnitz** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/zuul-opensource-repository-check/74/ : SUCCESS in 10s
- https://ci.intgdc.com/job/zuul-tox-zuul-el7/86/ : FAILURE in 10m 29s

**Comment by janpriessnitz** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/zuul-opensource-repository-check/75/ : SUCCESS in 9s
- https://ci.intgdc.com/job/zuul-tox-zuul-el7/87/ : SUCCESS in 7m 28s

File: tests/test_github.py [hunk 1]
```diff
@@ -542,3 +542,12 @@ def test_custom_github_status_name(self):
...
         self.assertIn('abcd', A.statuses)
+
+    def test_base_changed_event(self):
+        "Tests that Zuul responds to base change in Github PR"
+        A = self.fake_github.openFakePullRequest('org/project', 'master', 'A')
+        self.fake_github.emitEvent(A.getPullRequestBaseChangedEvent())
+        self.waitUntilSettled()
+
+        self.assertEqual(len(self.history), 3)
+        self.assertEqual(A.statuses['check']['state'], 'success')
```
Comment (hrubi) [inline tests/test_github.py]: Add a newline on the end of the file.

File: tests/base.py [hunk 2]
```diff
@@ -535,6 +535,13 @@ def getPullRequestReopenedEvent(self):
     def getPullRequestClosedEvent(self):
         return self._getPullRequestEvent('closed')
 
+    def getPullRequestBaseChangedEvent(self):
+        name, data = self._getPullRequestEvent('edited')
+        data['changes'] = {
+            'base': 'new_base_branch'
+        }
+        return (name, data)
```
Comment (hrubi) [inline tests/base.py]: Can you please link me to the api doc? Is this where you get it from? https://developer.github.com/v3/activity/events/types/#pullrequestevent Or did you have to try it out to find out what GH sends? The GitHub API seems to get richer and richer and the documentation IMO lacks behind a bit..

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/zuul-opensource-repository-check/76/ : SUCCESS in 8s
- https://ci.intgdc.com/job/zuul-tox-zuul-el7/88/ : FAILURE in 11m 06s

**Comment by janpriessnitz** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/zuul-opensource-repository-check/81/ : SUCCESS in 13s
- https://ci.intgdc.com/job/zuul-tox-zuul-el7/93/ : FAILURE in 11m 24s

**Comment by janpriessnitz** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/zuul-opensource-repository-check/82/ : SUCCESS in 8s
- https://ci.intgdc.com/job/zuul-tox-zuul-el7/94/ : SUCCESS in 7m 47s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://ci.intgdc.com/job/zuul-merge-compliance-check/11/ : SUCCESS in 19s
- https://ci.intgdc.com/job/zuul-tox-zuul-el7/96/ : SUCCESS in 7m 51s

**Pull request merged**